kms/kms.rb

```diff
@@ -0,0 +1,392 @@
+  resource = "projects/#{project_id}/locations/#{location}/" +
+           "keyRings/#{key_ring_id}/cryptoKeys/#{crypto_key}/" +
+           "cryptoKeyVersions/#{version}"
+
+  # Destroy specific version of the crypto key
+  kms_client.destroy_crypto_key_version(resource,
+      Cloudkms::DestroyCryptoKeyVersionRequest.new)
+
+  puts "Destroyed version #{version} of #{crypto_key}"
+  # [END kms_destroy_cryptokey_version]
+end
+
+$add_member_to_cryptokey_policy = -> (project_id:, key_ring_id:, crypto_key:,
+                                      member:, role:, location:) do
+  # [START kms_add_member_to_cryptokey_policy]
+  # project_id  = "Your Google Cloud project ID"
+  # key_ring_id = "The id of the new KeyRing"
+  # crypto_key  = "Name of the cryptoKey"
+  # member      = "Member to add to cryptoKey policy"
+  # role        = "Role assignment for new member"
+  # location    = "The location of the new KeyRing"
+
+  require "google/apis/cloudkms_v1beta1"
+
+  # Instantiate the client, authenticate with specified scope
+  Cloudkms = Google::Apis::CloudkmsV1beta1
+  kms_client = Cloudkms::CloudKMSService.new
+  kms_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the KeyRing
+  resource = "projects/#{project_id}/locations/#{location}/" +
+             "keyRings/#{key_ring_id}/cryptoKeys/#{crypto_key}"
+
+  # Get the current IAM policy
+  policy = kms_client.get_project_location_key_ring_crypto_key_iam_policy resource
+
+  # Add new member to current bindings
+  policy.bindings ||= []
+  policy.bindings << Cloudkms::Binding.new(members: [member], role: role)
+
+  # Update IAM policy
+  policy_request = Cloudkms::SetIamPolicyRequest.new(policy: policy)
+  kms_client.set_crypto_key_iam_policy resource, policy_request
+
+  puts "Member #{member} added to policy for " +
+       "key #{crypto_key} in keyring #{key_ring_id}"
+  # [END kms_add_member_to_cryptokey_policy]
+end
+
+$get_keyring_policy = -> (project_id:, key_ring_id:, location:) do
+  # [START kms_get_keyring_policy]
+  # project_id  = "Your Google Cloud project ID"
+  # key_ring_id = "The id of the new KeyRing"
+  # location    = "The location of the new KeyRing"
+
+  require "google/apis/cloudkms_v1beta1"
+
+  # Instantiate the client, authenticate with specified scope
+  Cloudkms = Google::Apis::CloudkmsV1beta1
+  kms_client = Cloudkms::CloudKMSService.new
+  kms_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the KeyRing
+  resource = "projects/#{project_id}/locations/#{location}/" +
+             "keyRings/#{key_ring_id}"
+
+  # Get the current IAM policy
+  policy = kms_client.get_project_location_key_ring_iam_policy resource
+
+  # Print role and associated members
+  unless policy.bindings.nil?
```

Comment:
> *Never, ever, ever use an **else** clause with an **unless** statement.*

 - *[Making sense with Ruby's "unless"](https://signalvnoise.com/posts/2699-making-sense-with-rubys-unless)*

😋 

It's [debated a little](http://stackoverflow.com/questions/38627935/do-we-use-else-with-unless-statement), but switch to an `if` conditional when you want an `else` clause.

```ruby
if policy.bindings
  # ...
else
  # ...
end
```

**#misc** man it makes me crazy that bindings can be nil instead of being empty...